examples/platforms/nrf528xx/src/radio.c

```diff
@@ -452,19 +452,35 @@ otError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
 
 otError otPlatRadioTransmit(otInstance *aInstance, otRadioFrame *aFrame)
 {
-    bool result = true;
+    bool    result = true;
+    otError error  = OT_ERROR_NONE;
 
     aFrame->mPsdu[-1] = aFrame->mLength;
 
     nrf_802154_channel_set(aFrame->mChannel);
 
-    if (aFrame->mInfo.mTxInfo.mCsmaCaEnabled)
+#if OPENTHREAD_CONFIG_CSL_USE_PLATFORM_DELAY_TRANSMIT
```

Comment:
Check OT version >= 1.2 instead here and return correct radio capability according to OT version.